stat11_backend/stat11/utils/match_team_list_data.py

```diff
@@ -0,0 +1,45 @@
+from stat11.models import Team, Player
+from stat11.serializers import TeamSerializer , PlayerNestedSerializer
+
+def get_match_team_list_data(match_id):
+    teamList = []
+    matchTeams = Team.objects.filter(match__id = match_id)
+    firstTeam = matchTeams[0]
+    secondTeam = matchTeams[1]
+    serializer1 = TeamSerializer(firstTeam)
+    data1 = serializer1.data
+    serializer2 = TeamSerializer(secondTeam)
+    data2 = serializer2.data
+    teamList.append(data1['players'])
+    teamList.append(data2['players'])
+    netTeam = []
+    for i,pl in enumerate(teamList):
+        team = []
+        for j,pid in enumerate(pl):
+            # print(pid)
+            player = Player.objects.all()
+            for k in player:
+                if(k.id == pid):
+                    data = k
```

Comment:
Rather than again and again querying for the complete player list and running a loop, just query directly as 'player = Player.objects.get(id=pid)' ..... Along with proper exception handling

Reply:
See where it's going wrong and use it the right way.......you can't query the whole database table again and again just for one object and that also in a third nested loop

Reply:
ok will look into this

Reply:
Resolved